pic-sure-resources/pic-sure-aggregate-data-sharing-resource/src/main/java/edu/harvard/hms/dbmi/avillach/AggregateDataSharingResourceRS.java

```diff
@@ -29,707 +24,809 @@
+            continuousCrossCounts.forEach((key, value) -> {
+                Map<String, Object> innerMap = new HashMap<>(value);
+                convertedContinuousCrossCount.put(key, innerMap);
+            });
+
+            obfuscatedCrossCount(generatedVariance, convertedContinuousCrossCount);
+            return objectMapper.writeValueAsString(convertedContinuousCrossCount);
+        }
+    }
+
+    /**
+     * This method handles the processing of categorical cross counts. It begins by determining whether the cross
+     * counts require obfuscation. This is accomplished by checking if any of the CROSS_COUNTS must be obfuscated.
+     * If obfuscation is required, the categorical cross counts will be obfuscated accordingly. Otherwise,
+     * if no obfuscation is needed, the method can simply return the categorical entity string.
+     *
+     * @param categoricalEntityString The categorical entity string
+     * @param crossCountEntityString  The cross count entity string
+     * @return String The processed categorical entity string
+     * @throws JsonProcessingException If there is an error processing the JSON
+     */
+    protected String processCategoricalCrossCounts(String categoricalEntityString, String crossCountEntityString) throws JsonProcessingException {
+        logger.info("Processing categorical cross counts");
+
+        if (categoricalEntityString == null || crossCountEntityString == null) {
+            return null;
+        }
+
+        Map<String, String> crossCounts = objectMapper.readValue(crossCountEntityString, new TypeReference<>() {});
+        int generatedVariance = this.generateVarianceWithCrossCounts(crossCounts);
+
+        boolean mustObfuscate = true;
+        if (!mustObfuscate) {
+            return categoricalEntityString;
+        }
+
+        // This might break in the object mapper. We need to test this.
+        Map<String, Map<String, Integer>> categoricalCrossCount = objectMapper.readValue(categoricalEntityString, new TypeReference<>() {});
+
+        if (categoricalCrossCount == null) {
+            return categoricalEntityString;
+        }
+
+        for (Map.Entry<String, Map<String, Integer>> entry : categoricalCrossCount.entrySet()) {
+            // skipKey is expecting an entrySet, so we need to convert the axisMap to an entrySet
+            if (skipKey(entry)) continue;
+
+            Map<String, Integer> axisMap = processResults(entry.getValue());
+            categoricalCrossCount.put(entry.getKey(), axisMap);
+        }
+
+        // Convert the categoricalCrossCount Map to a map<String, Map<String, Object>>
+        Map<String, Map<String, Object>> convertedCategoricalCrossCount = new HashMap<>();
+        categoricalCrossCount.forEach((key, value) -> {
+            Map<String, Object> innerMap = new HashMap<>(value);
+            convertedCategoricalCrossCount.put(key, innerMap);
+        });
```

Comment:
Because the entire file was formatted I am pointing out the modified code. This was added to accomplish the desired results.